src/main/java/com/melon/portfoliomanager/services/NotificationService.java

```diff
@@ -0,0 +1,12 @@
+package com.melon.portfoliomanager.services;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+@Service
+public class NotificationService {
+    public void notifyUser(Map<String, Double> companyStocks) {
```

Comment:
Is NotificationService a Kafka producer or meant to actually send out notifications to the user? Won't the actual user notifications (email) be handled in a separate REST API?


If the Kafka message publishing will be done here, then this service shouldn't be aware of users and company stocks. All messages going through Kafka should be serialized first before they reach this point, so there's no need to have separate methods for each data type (e.g. company stocks). The Kafka producer doesn't really care about users either, it just publishes messages to some topic, then assuming we have a separate notifications-service API it will subscribe to that topic and consume messages as they come, deserialize them and do the actual user notifications.

It's fine for now, just be aware to do proper separation of concerns when adding the Kafka stuff later.

Reply:
```NotificationService``` is just a placeholder reminding us that at this point (or somewhere near it) we will need to send messages in Kafka (not the actual messages to the end users, i.e. not the actual email notifications). This is how I imagined things will be done - there are separate tasks on our board in Trello (https://trello.com/b/NQT9WNVQ/portfolio-manager) inside ```To Do``` column for setting up Kafka and start sending messages to Kafka.

Reply:
About the implementation details of message sending to the message broker - they will be handled in another task.
(the serialization of the messages may be done outside of the ```NotificationService```, yeah - will think about that when the actual work on pushing messages to Kafka start)

Reply:
I don't think that adding empty classes as reminders is a good practice. If the work is described in a separate ticket then there's no need to add "reminder code" as well.